### PR TITLE
Update Silverfrost compiler description.

### LINKED ISSUE #301: Update Silverfrost compiler description
Based on the discussion [here](https://fortran-lang.discourse.group/t/silverfrost-compiler-adds-some-f2003-and-f2008-features/1584), I suggest editing the Silverfrost compiler description by appending the sentences, 

Some features of Fortran 2003 and 2008 have been [added](https://www.silverfrost.com/19/ftn95/support/ftn95_revision_history.aspx). There is a free personal edition.

### PR DESCRIPTION
**Pull request by brocolis**
Author: Beliavsky; Closes #301.

### CODE REVIEW

**Review by certik — APPROVED** [review]

File: compilers.md [hunk 1]
```diff
@@ -164,6 +164,8 @@ Microsoft .NET. FTN95 ships with the world's best runtime checking and a great
 range of supporting software. All standard and many vendor-specific legacy
 language features are supported, so that Fortran projects may be any combination
 of Fortran 77, Fortran 90 and Fortran 95.
+Some features of Fortran 2003 and 2008 have been [added](https://www.silverfrost.com/19/ftn95/support/ftn95_revision_history.aspx).
+There is a free personal edition.
```
Comment (awvwgk) [inline compilers.md:168]: Could you add the information about the supported platforms and architectures if those are available?

Comment (brocolis) [reply compilers.md:168]: According to this page https://www.silverfrost.com/33/ftn95/faq.aspx

> Silverfrost Fortran runs on both 32 and 64-bit Windows. 
> …
> There is only one version of Silverfrost Fortran and it runs on 64-bit Windows just as happily as 32-bit.
> …
> If your programs need huge amounts of memory then turning on 64-bit mode will help you.

I will include the first sentence to this PR.

File: compilers.md [hunk 2]
```diff
@@ -164,6 +164,8 @@ Microsoft .NET. FTN95 ships with the world's best runtime checking and a great
 range of supporting software. All standard and many vendor-specific legacy
 language features are supported, so that Fortran projects may be any combination
 of Fortran 77, Fortran 90 and Fortran 95.
+Some features of Fortran 2003 and 2008 have been [added](https://www.silverfrost.com/19/ftn95/support/ftn95_revision_history.aspx).
+Silverfrost Fortran runs on both 32 and 64-bit Windows. There is a free personal edition.
```
Comment (awvwgk) [inline compilers.md:168]: To be consistent with other compiler descriptions. x86 support is implied in a way.
```suggestion
Silverfrost Fortran runs on Windows / x86_64. There is a free personal edition.
```

**Comment by brocolis** [comment]
Thank you for reviewing it.

**Review by awvwgk — APPROVED** [review]

**Review by milancurcic — APPROVED** [review]
Thank you!

**Pull request merged**